src/common/pool_map.c

```diff
@@ -1323,17 +1338,17 @@ pool_map_merge(struct pool_map *map, uint32_t version,
 			/* new buffer may have changes for this domain */
 			if (sdom->do_children != NULL) {
 				struct pool_domain *child = addr;
-				struct pool_comp_cntr	s_cntr;
+				uint32_t	tgt_nr;
 
-				pool_tree_count(sdom, &s_cntr);
```

Comment:
sorry, why the pool_tree_count() "might count the target count under other domain"? seems pool_tree_count() only for the case sdom->do_children == NULL it will count all tgts for the same level domains? here for the case of "sdom->do_children != NULL)" it won't affect it?

Reply:
Here it suppose to calculate all targets belong to this domain.   But pool_tree_count() actually calculate all targets from this domain.  For example if the tree structure is 

domain1, domain2,domain3, targets0---target7,target8-target15, target16-target-23

so if it suppose to get all targets of domain1(targets0--target7), then it can not use pool_tree_count, which will get all 24 targets.  instead  it should use the new function.

Reply:
i see, thanks. checked again the code it is same for either != NULL or not.